internal/services/exchange.service.go

```diff
@@ -8,81 +8,85 @@ import (
 	"momentum-go-server/internal/utils"
 	"net/http"
 	"os"
-	"slices"
-	"strings"
 	"time"
 
 	_ "github.com/joho/godotenv/autoload"
 )
 
 var currentTime = time.Now()
 var yesterday = currentTime.AddDate(0, 0, -1)
-var symbolsArr = []string{"AUD", "BRL", "BTC", "CAD", "CLP", "CNY", "CZK", "EGP", "EUR", "GBP", "HKD", "INR", "JPY", "KRW", "LTL", "LVL", "TRY", "USD", "XAG", "XAU"}
 
-func getNBUData(date string) []models.NBU {
+// var symbolsArr = []string{"AUD", "BRL", "EGP", "CAD", "CLP", "CNY", "CZK", "EGP", "EUR", "GBP", "HKD", "INR", "JPY", "KRW", "LTL", "LVL", "TRY", "USD", "XAG", "XAU", "UAH", "PLN"}
+
+func getNBUData(date, symbol string) models.NBU {
 	var response []models.NBU
+	var currentRate float64
 
-	apiURL := fmt.Sprintf("https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?date=%s&json", date)
+	apiURL := fmt.Sprintf("https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?valcode=%s&date=%s&json", symbol, date)
 
 	resp, err := http.Get(apiURL)
 
 	if err != nil {
 		utils.ErrorLogger.Println("Error creating HTTP request:", err)
-		return response
+		return response[0]
```

Comment:
Check it before if array isn't empty

Reply:
Fixed: Default structure is now returned on error. There is also a check an empty response.